src/main/java/org/mozilla/zest/impl/ZestHttpClient.java

```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.zest.impl;
+
+import org.mozilla.zest.core.v1.ZestAuthentication;
+import org.mozilla.zest.core.v1.ZestRequest;
+import org.mozilla.zest.core.v1.ZestResponse;
+
+import java.io.IOException;
+
+public interface ZestHttpClient{
+
+    void setAuthentication(ZestAuthentication zestAuthentication);
```

Comment:
- Maybe `addAuthentication`? "set" implies the previous authentication is replaced, while it might not be the case (it's cumulative if targeting different hosts).
 - Any special reason to use `ZestAuthentication` instead of `ZestHttpAuthentication`?

Reply:
#Any special reason to use ZestAuthentication instead of ZestHttpAuthentication?
For the case that there is another Authentication i.e. jwt bearer token in the header

Reply:
OK, that's what I suspected, but more work would be required to support other authentication methods ;)